app/routes.py

```diff
@@ -1 +1,280 @@
+
+    db.session.commit()
+
+    post_to_slack(task)
+
+    return make_response({"task": to_dict(task)})
+
+#------------MARK TASK INCOMPLETE-----------#
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def uncheck_task(task_id):
+    task = validate_item(Task,task_id)
+    task.completed_at=None
+
+    db.session.commit()
+
+    return make_response({"task": to_dict(task)})
+
+#---------------------------------------------------------------#
+#-------------------------GOAL ROUTES---------------------------#
+#---------------------------------------------------------------#
+
+#-------------GET ALL GOALS------------#
+@goal_bp.route("", methods =["GET"])
+def get_goals():
+    goals = query_filter(Goal)
+    goals_response = [to_dict(goal) for goal in goals]
+
+    return make_response(jsonify(goals_response), 200)
+
+#-------------MAKE NEW GOAL-------------#
+@goal_bp.route("", methods =["POST"])
+def make_goal():
+    request_body = request.get_json()
+    new_goal = validate_new_item(Goal,request_body)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response({"goal":to_dict(new_goal)}, 201)
+
+#-------------GET SINGLE GOAL------------#
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def one_goal(goal_id):
+    goal = validate_item(Goal,goal_id)
+
+    return make_response(dict(goal=to_dict(goal)), 200)
+
+#-------------EDIT SINGLE GOAL-------------#
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def edit_goal(goal_id):
+    goal = validate_item(Goal,goal_id)
+    request_body = request.get_json()
+
+    if "title" in request_body:
+        goal.title=request_body["title"]
+    if "description" in request_body:
+        goal.description=request_body["description"]
+
+    db.session.commit()
+
+    return(make_response({"goal":to_dict(goal)}, 200))
+
+#-------------DELETE SINGLE GOAL-------------#
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_item(Goal,goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'},200)
+
+#---------------------------------------------------------------#
+#-----------------------NESTED ROUTES---------------------------#
+#---------------------------------------------------------------#
+
+#-------------GET ALL GOAL'S TASKs-------------#
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_tasks_for_goal(goal_id):
+    goal = validate_item(Goal,goal_id)
+    goal_response = to_dict(goal,list_tasks=True)
+
+    return make_response(jsonify(goal_response), 200)
+
+#---------------ADD TASKS TO GOAL--------------#
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def add_tasks(goal_id):
+    goal = validate_item(Goal,goal_id)
+    request_body = request.get_json()
+    
+    if "task_ids" in request_body:
+        task_list = [task_to_goal(id,goal) for id in request_body["task_ids"]]
```

Comment:
I like the spirit of this function - you want to update each task from the list of IDs passed to this endpoint with the goal ID and then end up with a list of those ids. Since we're passed a list of task IDs already, can we not simply return that in the response and skip the additional step of returning an ID for each task that's being updated in the `task_to_goal` function? Then the function could be named something more specific, like `update_task_with_goal_id or something, and we would also have a function that's a bit more single-purpose.